control-plane/rest/src/lib.rs

```diff
@@ -148,51 +161,160 @@ impl ActixRestClient {
             self.client.delete(uri.clone()).send().await
         };
 
-        let rest_response = result.map_err(|error| {
-            anyhow::anyhow!(
-                "Failed to delete uri '{}' from rest, err={:?}",
-                uri,
-                error
-            )
+        let rest_response = result.context(Send {
+            details: format!("Failed to delete uri {}", uri),
         })?;
 
         Self::rest_result(rest_response).await
     }
 
     async fn rest_vec_result<S, R>(
         mut rest_response: ClientResponse<S>,
-    ) -> anyhow::Result<Vec<R>>
+    ) -> ClientResult<Vec<R>>
     where
         S: Stream<Item = Result<Bytes, PayloadError>> + Unpin,
         for<'de> R: Deserialize<'de>,
     {
-        let rest_body = rest_response.body().await?;
-        if rest_response.status().is_success() {
-            match serde_json::from_slice(&rest_body) {
-                Ok(result) => Ok(result),
-                Err(_) => Ok(vec![serde_json::from_slice::<R>(&rest_body)?]),
+        let status = rest_response.status();
+        let headers = rest_response.headers().clone();
+        let head = || {
+            let mut head = ResponseHead::new(status);
+            head.headers = headers.clone();
+            head
+        };
+        let body = rest_response.body().await.map_err(|_| {
```

Comment:
Should we include the error being mapped?

Reply:
yeah, I've changed all the map's to context now :+1: